pkg/controller/jobframework/reconciler.go

```diff
@@ -164,6 +182,32 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// 1.1 If the workload is pending deletion, suspend the job if needed
+	// and drop the finalizer.
+	if wl != nil && !wl.DeletionTimestamp.IsZero() {
+		log.V(2).Info("The workload is marked for deletion")
+		if !job.IsSuspended() {
+			err := r.stopJob(ctx, job, object, wl, "Workload is deleted")
+			if err != nil {
+				log.Error(err, "Suspending job with deleted workload")
+			}
+			return ctrl.Result{}, err
+		} else {
+			// make sure the finalizer is removed
+			err := r.setFinalizer(ctx, wl, false)
+			return ctrl.Result{}, err
+		}
+	}
+
+	// 1.2 if a job stop failure is suspected
+	if suspectJobStopFailure(job, wl) {
```

Comment:
This looks rather suspicious. First, why do we need this , and do we need this given https://github.com/kubernetes-sigs/kueue/pull/1005. Second, if really needed add a comment to explain a scenario in which this is needed.